### PR TITLE
[WIP] Feature/protocol generator tests generation

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

scaffold protocol tests on generation

## Fixes

https://github.com/valory-xyz/open-aea/issues/530

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Comment by solarw** [comment]
@DavidMinarsch @angrybayblade 

this PR contains auto generated tests for protocols:
test_messages and test_dialogues

files are not edited manually, require adjust custom data types. generic types values added automatically.

File: packages/fetchai/protocols/default/tests/test_dialogues.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    DefaultDialogue,
+    DefaultDialogues,
+)
+from packages.fetchai.protocols.default.message import DefaultMessage
+
+
+class TestDialoguesDefault(BaseProtocolDialoguesTestCase):
+    """Test for the 'default' protocol dialogues."""
+
+    __test__ = True
```
Comment (angrybayblade) [inline packages/fetchai/protocols/default/tests/test_dialogues.py:35]: I thought we didn't required this

File: aea/protocols/generator/base.py [hunk 2]
```diff
@@ -2146,6 +2148,304 @@ def generate_full_mode(self, language: str) -> Optional[str]:
                 full_mode_output = incomplete_generation_warning_msg
         return full_mode_output
 
+    def _generate_tests(self):
+        tests_dir = str(Path(self.path_to_generated_protocol_package) / "tests")
+        os.makedirs(tests_dir, exist_ok=True)
+
+        TESTS_MESSAGES_DOT_PY_FILE_NAME = "test_messages.py"
```
Comment (angrybayblade) [inline aea/protocols/generator/base.py:2155]: Should be at top

File: aea/protocols/generator/base.py [hunk 3]
```diff
@@ -2146,6 +2148,304 @@ def generate_full_mode(self, language: str) -> Optional[str]:
...
+        _create_protocol_file(
+            tests_dir,
+            TESTS_MESSAGES_DOT_PY_FILE_NAME,
+            self._test_messages_file_str(),
+        )
+
+        TESTS_DIALOGUES_DOT_PY_FILE_NAME = "test_dialogues.py"
```
Comment (angrybayblade) [inline aea/protocols/generator/base.py:2162]: Same as above

**Review by angrybayblade — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**